### DISCUSSION TITLE
Cannot delete budibase table row

### DISCUSSION THREAD

**Discussion by applicationuser** [Budibase/budibase#8973] [Help]
Hello,

Cannot delete budibase table row in Data mode, On delete get success confirmation but row not deleted in table.

Table has relationship column to another table and rows in that table are deleted related to this row.

No errors in browser console/ POST api response success.

Budibase : Cloud
App Version:  2.1.42
App published 

Thanks!

**Comment by oitkenmen** [comment]
It happens sometimes. Try to change the tab or quit the app and then enter it again. The database cannot delete/update the row if this will create an inconsistency in the data. Try to check all of the data in related tables. Especially if you have formulas.

**Reply by applicationuser** [reply]
Thanks @berkimbayev 
Checked all tables , no dependent rows. still cannot delete . any thing else i can check?

**Comment by melohagan** [comment]
Hey @applicationuser 

Did you get this resolved? If you're still having issues maybe you could send your app export to support@budibase.com and I can have a look.

**Reply by applicationuser** [reply]
Hi @melohagan 

Thank you for looking into this, We are still having issue. App export sent to support email

Thanks

**Reply by melohagan** [reply]
Hey @applicationuser None of the tables actually had any rows in the app export you sent through.

It looks to be a fairly complex setup of relationships, so having some test data and which row from which table you're trying to delete would help a lot.

**Reply by applicationuser** [reply]
Hello @melohagan 

i have sent another email with app export including data and details about which table has issues.

**Reply by melohagan** [reply]
Hey @applicationuser 

Thank you very much for the detailed email. I was able to determine the reason you couldn't delete the rows is because there were many conflicts in the DB for those rows you were trying to delete.

I'm not sure how that happened, but there is a guide here on how to resolve conflicts: https://github.com/Budibase/budibase/discussions/7777#discussioncomment-3732390

In this case however, the conflict wasn't in the *design/database* but in the rows themselves.
For each row you want to delete, you'll need to get the row id, search for it, then resolve conflicts.

**Getting the row id**
<img width="527" alt="Screenshot 2022-12-20 at 19 55 42" src="https://user-images.githubusercontent.com/101575380/208755008-339a5aa7-0abe-4413-b5bf-b4998a39a6e7.png">

Click on the eye icon to show autocolumns. Then click the overlapping squares to copy the row id to clipboard.

**Finding the document**
After logging into the couchdb console, find and click on your app id. This is found in the URL.
<img width="291" alt="Screenshot 2022-12-20 at 19 56 41" src="https://user-images.githubusercontent.com/101575380/208755169-5b17aca7-8598-4a4e-9a4e-5f7323a59316.png">

Next paste the copied ID into the *Document ID* search box
<img width="1141" alt="Screenshot 2022-12-20 at 19 58 01" src="https://user-images.githubusercontent.com/101575380/208755394-daa2b94f-bbdc-4683-a943-004c6f0d9948.png">

It should present a link to click on:
<img width="1141" alt="Screenshot 2022-12-20 at 19 58 38" src="https://user-images.githubusercontent.com/101575380/208755507-b6e75414-42ab-4f0a-b139-e50746d8b9e7.png">

Finally follow the guide I linked from **Step 5** to resolve the conflicts. You should then be able to delete the rows within the **Data** section of Budibase (I tested this to be the case).

**Reply by applicationuser** [reply]
Thank you @melohagan  for looking into this.  We are using budibase cloud . how to access and delete data from couchdb ? 

also can you check automation , one of them 'GenerateQ.... Automation' using create row step in loop and its only looping to max 200 rows. is that a limitation in automation or something happening with data conflict in the app ?

**Reply by melohagan** [reply]
Hey @applicationuser 

The max 200 rows is a hard limit for automation looping.

We have resolved the document conflicts in your app within the cloud environment if you want to verify that the problem has been resolved for you. 
If the issue resurfaces, please let us know as soon as possible, as it will help us to determine what caused the conflict in the first place.

Also you are using quite a lot of relationships within your app. It may be more maintainable to use a SQL database.

**Reply by applicationuser** [reply]
Hi @melohagan 

Thank for fixing the issue, now we are able to delete those rows. 

will let you know if the issue resurfaces. 

Ok. I'll take a look at the relationship - we only have 3 relationship tables and formula columns based on them.

do you have any suggestion what option we would have if we remove relationship ?

Thanks again for looking into this.
